src/resource/factory/LightFactory.cpp

```diff
@@ -0,0 +1,15 @@
+#include "resource/factory/LightFactory.h"
+#include "resource/type/Light.h"
+
+std::shared_ptr<Ship::IResource> LUS::ResourceFactoryBinaryLightV0::ReadResource(std::shared_ptr<Ship::File> file) {
+    if (!FileHasValidFormatAndReader(file)) {
+        return nullptr;
+    }
+
+    std::shared_ptr<Light> light = std::make_shared<Light>(file->InitData);
+    auto reader = std::get<std::shared_ptr<Ship::BinaryReader>>(file->Reader);
+
+    reader->Read((char*)light->GetPointer(), sizeof(LightEntry));
```

Comment:
this part could be spit up into a bunch of `Read{type}` calls to put the data in the appropriate places and also "read away" the padding without storing it anywhere.

Reply:
I tried, this works better, we had a full conversation about it on the Mega's port channel